modules/standard/GMP.chpl

```diff
@@ -109,11 +109,51 @@ module GMP {
   use SysError;
   use BigInteger;
 
+  require "GMPHelper/chplgmp.h";
+
+  export proc chpl_gmp_alloc(size:size_t) : c_void_ptr {
+    pragma "insert line file info"
+    extern proc chpl_mem_alloc(size:size_t, md:chpl_mem_descInt_t) : c_void_ptr;
+    extern const CHPL_RT_MD_GMP:chpl_mem_descInt_t;
+    return chpl_mem_alloc(size, CHPL_RT_MD_GMP);
```

Comment:
I wonder if we can just use `chpl_here_alloc()` and friends now instead of duplicating the externs. I won't hold up the PR over that, but it might be a nice cleanup